pontoon/base/views.py

```diff
@@ -414,12 +414,17 @@ def get_translation_history(request):
     for t in translations:
         u = t.user or User(username="Imported", first_name="Imported", email="imported")
         translation_dict = t.serialize()
+        if t.user:
+            user_role = User.locale_role(t.user, locale)
+        else:
+            user_role = ""
         translation_dict.update(
             {
                 "user": u.name_or_email,
                 "uid": u.id,
                 "username": u.username,
                 "user_gravatar_url_small": u.gravatar_url(88),
+                "user_role": user_role,
```

Comment:
I believe the following should be the same (and then you can drop the new code above):
```suggestion
                "user_role": u.locale_role,
```

However, `User.locale_role()` is not exactly what we need here according to the spec. We also need the "New" icon and don't need the "System User" and "Contributor" roles.